src/test/cpp/blaze_util_test.cc

```diff
@@ -234,4 +234,88 @@ TEST_F(BlazeUtilTest, TestSearchUnaryCommandOptionWarnsAboutDuplicates) {
   }
 }
 
+void assert_equal_vector_char_pointer(std::vector<std::string> expected, std::vector<std::string> actual) {
+  ASSERT_EQ(actual.size(), expected.size()) << "Vectors expected and actual are of unequal length";
+
+  for (int i = 0; i < actual.size(); ++i) {
+    ASSERT_EQ(actual[i], expected[i]) << "Vectors expected and actual differ at index " << i;
+  }
+}
+
+TEST_F(BlazeUtilTest, TestSearchAllUnaryForEmpty) {
+  assert_equal_vector_char_pointer({}, GetAllUnaryOptionValues({"bazel", "build", ":target"}, ""));
+}
+
+TEST_F(BlazeUtilTest, TestSearchAllUnaryFlagNotPresent) {
+  assert_equal_vector_char_pointer({}, GetAllUnaryOptionValues({"bazel", "build", ":target"}, "--flag"));
+}
+
+TEST_F(BlazeUtilTest, TestSearchAllUnaryStartupOptionWithEquals) {
+  assert_equal_vector_char_pointer({"value"}, GetAllUnaryOptionValues({"bazel", "--flag=value", "build", ":target"}, "--flag"));
+}
+
+TEST_F(BlazeUtilTest, TestSearchAllUnaryStartupOptionWithEquals2) {
+  assert_equal_vector_char_pointer({"value1", "value2"}, GetAllUnaryOptionValues({"bazel", "--flag=value1", "--flag=value2", "build", ":target"}, "--flag"));
+}
+
+TEST_F(BlazeUtilTest, TestSearchAllUnaryStartupOptionWithRepeatingFlag) {
+  assert_equal_vector_char_pointer({"--flag"}, GetAllUnaryOptionValues({"bazel", "--flag", "--flag", "value1", "build", ":target"}, "--flag"));
+}
+
+TEST_F(BlazeUtilTest, TestSearchAllUnaryStartupOptionWithRepeatingFlagOptionsOnly) {
+  assert_equal_vector_char_pointer({"--flag"}, GetAllUnaryOptionValues({"bazel", "--flag", "--flag", "value1"}, "--flag"));
+}
+
+TEST_F(BlazeUtilTest, TestSearchAllUnaryStartupOptionWithRepeatingFlagOptionsOnlyMulti) {
+  assert_equal_vector_char_pointer({"--flag", "value1"}, GetAllUnaryOptionValues({"bazel", "--flag", "--flag", "--flag", "value1"}, "--flag"));
+}
+
+TEST_F(BlazeUtilTest, TestSearchAllUnaryStartupOptionWithRepeatingFlagOptionsOnlyMultiWithEquals) {
```

Comment:
Lines should be <=80 cols. The import process auto-formats to get most lines there, but it's having a hard time with this one because of the length of the test name. Can you shorten it a bit? Just one or two cols over right now.

Reply:
Manually corrected.

> The import process auto-formats

What do you mean by the import process? Is there an auto formatter I can run in this repo?

Reply:
The process of merging the change. I'm not sure on the formatter question.

Reply:
Thanks!